### ISSUE TITLE
Batching tutorial with the Character-Level Names classification

### ISSUE THREAD

**Issue by ngarneau** [spro/practical-pytorch#67]

**Comment by hunkim** [comment]
@ngarneau Wow this demo (matrix visualization) is amazing. I have two questions for you.

1. I am making slides for RNN (bit.ly/PyTorchZeroAll). May I use your matrix visualization in my slides (with your name credit?)

2. I am also working on a simple seq2seq model for a tutorial based on the code in this repos (https://github.com/hunkim/PyTorchZeroToAll/blob/master/13_1_seq2seq.py). Are you interested in visualizing the metrics used in the seq2seq code?

Once again, it's really amazing to see your wonderful visualization.

**Comment by ngarneau** [comment]
Hey @hunkim,
Thanks for your comment and please, use any of my content within your slides I'd be more than happy :)
It would be a pleasure for me to design visualizations for your seq2seq model. It may take a couple of days since I haven't been through one thoroughly yet but it should not be that long.
By the way, your slides are great! I think we are on the same page on "visualizing what the code does" exactly.
Many thanks,
Nicolas

**Comment by hunkim** [comment]
@ngarneau Take your time. I'll finalize my slides and video lectures next week. I really appreciate it.

**Comment by hunkim** [comment]
@ngarneau Do you know how to use pack-pad for decoder? In the example in this repos, he only used pack-pad in the encoder, but the decoder is just batch based loop.

**Comment by SnowIsWhite** [comment]
In batched-seq2seq, he used pack padded sequences in decoder as well. He
then explioted codes in masked crossentropy.py to calculate the loss.

**Comment by ngarneau** [comment]
@hunkim actually no but I'll give it a look this weekend!

**Comment by hunkim** [comment]
FYI, I also added data parallel. Please check it out at https://github.com/hunkim/PyTorchZeroToAll/blob/master/12_4_name_classify.py.

**Comment by ngarneau** [comment]
That is really cool! Thanks :)

**Comment by hunkim** [comment]
@ngarneau However, for this small dataset, it actually takes more time. :-)
